packages/playwright-core/src/server/android/android.ts

```diff
@@ -264,9 +264,13 @@ export class AndroidDevice extends SdkObject {
     debug('pw:android')('Force-stopping', pkg);
     await this._backend.runCommand(`shell:am force-stop ${pkg}`);
     const socketName = isUnderTest() ? 'webview_devtools_remote_playwright_test' : ('playwright-' + createGuid());
-    const commandLine = this._defaultArgs(options, socketName).join(' ');
+    const commandLine = this._defaultArgs(options, socketName).map(arg => {
+      if (!arg.includes('\'') && !arg.includes('"'))
+        return `"${arg}"`;
+      return arg;
+    }).join(' ');
     debug('pw:android')('Starting', pkg, commandLine);
-    await this._backend.runCommand(`shell:echo "${commandLine}" > /data/local/tmp/chrome-command-line`);
+    await this._backend.runCommand(`shell:echo "${commandLine.replace(/"/g, '\\"')}" > /data/local/tmp/chrome-command-line`);
```

Comment:
You seem to be double-escaping here?

Reply:
We need to bash escape in the end, since we `echo` so quotes would not have ended up inside the `/data/local/tmp/chrome-command-line`.

Reply:
using base64 now and added a comment.